src/amphtml-email/20_Samples/Bookmark_Article_Email.html

```diff
@@ -0,0 +1,178 @@
+<!---{
+ "preview": "amp4email",
+ "skipValidation": "true",
+ "draft": "true"
+}--->
+<!--
+## Introduction
+
+This sample showcases how to use AMPHTML email to create an email with a
+personalized digest of articles, giving the user an ability to bookmark some of
+them for later reading.
+-->
+
+<!-- -->
+<!-- Doctype declaration is required. -->
+<!doctype html>
+<!-- This tells everyone that this is an AMPHTML email. `<html amp4email>` works too. -->
+<html ⚡4email>
+<!-- ## Head -->
+<!-- -->
+<head>
+  <!-- The charset definition must be the first child of the `<head>` tag. -->
+  <meta charset="utf-8">
+  <!-- The AMP runtime.-->
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+  <!--
+    The AMPHTML Email boilerplate.
+  -->
+  <style amp4email-boilerplate>body{visibility:hidden}</style>
+  <title>AMPHTML Email - Article List</title>
+  <!-- ## Setup -->
+  <!--
+    Additionally used AMP components must be imported in the header. We use
+    `amp-form` to submit bookmark requests to the server.
+  -->
+  <script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>
+  <!--
+    We use `amp-mustache` to render the result of a form submission.
+  -->
+  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
+  <style amp-custom>
+    .items {
+      list-style-type: none;
+      margin: 0;
+      padding: 0;
+    }
+
+    .item {
+      border-bottom: 1px solid #e6e6e6;
+      display: flex;
+      padding: 0 1rem;
+    }
+
+    .item > .link {
+      padding: 1.5rem 0.5rem;
+      flex-grow: 1;
+    }
+
+    .bookmark {
+      margin: auto 1rem;
+    }
+
+    .bookmark .submit-button {
+      float: right;
+    }
+
+    .bookmark form.amp-form-submitting button,
+    .bookmark form:not(.amp-form-submitting) div.submitting,
+    .bookmark form.amp-form-submit-success button {
+      display: none;
+    }
+
+    .bookmark form div[submit-error] {
+      margin-top: .4em;
+      clear: right;
+    }
+  </style>
+</head>
+<!-- ## Body -->
+<!-- -->
+<body>
+  <!--
+    Each list item has a normal link as well as an `amp-form` which is only used for sending an
```

Comment:
This sample has almost 100 lines of code and a single line of documentation. This is a bit hard to comprehend - in particular because the same code is repeated multiple times. Can you make this more focused?

Reply:
Any suggestions?

My aim is to make it clear the list itself is statically generated, with the `amp-form` being the only "special" part of it. I can drop it to two list items, but I'm not sure what I may do beyond that.

Reply:
One possibility would be to use amp-list. This has the further advantage that the content is always fresh (e.g. only the latest articles).